extension/memorylimiterextension/memorylimiter.go

```diff
@@ -12,29 +12,40 @@ import (
 	"go.opentelemetry.io/collector/internal/memorylimiter"
 )
 
-type memoryLimiterExtension struct {
-	memLimiter *memorylimiter.MemoryLimiter
+type MemoryLimiterExtension struct {
```

Comment:
@dmitryax @mx-psi I have some concerns with a component being used in a config package from other components to use. Having components depend on another component directly feels risky in a composable system. Could the necessary parts in `extension/memorylimiterextension` be moved to a shared location for the extension and `confighttp` to use?

Reply:
let me look through this again to detach coupling here.